src/morgan/morgan.middleware.ts

```diff
@@ -2,27 +2,62 @@ import * as morgan from "morgan";
 import { MORGAN_FORMAT_STRING } from "./morgan.constants";
 import { NestjsWinstonLoggerService } from "../nestjs-winston-logger.service";
 
+interface MorganOptions {
+  skip?: (request: Request, response: Response) => boolean;
+  immediate?: boolean;
+  format?: string;
+}
+
+const normalizeOptions = (
+  options: string | MorganOptions,
+  defaultFormat: string,
+): MorganOptions => {
+  if (!options) {
+    return { format: defaultFormat };
+  }
+
+  if (typeof options === "string") {
+    return { format: options };
+  }
+
+  return {
+    ...options,
+    format: options.format ?? defaultFormat,
+  };
+};
+
 export const morganRequestLogger = (
   logger: NestjsWinstonLoggerService,
-  morganFormatString: string = MORGAN_FORMAT_STRING.REQUEST,
-) =>
-  morgan(morganFormatString, {
-    immediate: true,
+  morganOptions: string | MorganOptions = MORGAN_FORMAT_STRING.REQUEST,
+) => {
+  const options = normalizeOptions(morganOptions, MORGAN_FORMAT_STRING.REQUEST);
+
+  return morgan(options.format, {
+    immediate: options?.immediate ?? true,
```

Comment:
Since we need to use { immediate: true } to control morgan for writing log line on request instead of response.
It looks a bit weird for letting users to set { immediate: false } in a function called  morganRequestLogger.

Reply:
k can remove this

Reply:
updated